sgtypes/src/channel_transaction.rs

```diff
@@ -420,19 +442,22 @@ impl CanonicalDeserialize for ChannelTransactionRequestPayload {
         let channel_txn_type = ChannelTransactionType::from_u32(decoded_txn_type);
         match channel_txn_type {
             Some(ChannelTransactionType::Offchain) => {
-                let witness = deserializer.decode_struct()?;
-                Ok(ChannelTransactionRequestPayload::Offchain(witness))
+                let hash_bytes = deserializer.decode_bytes()?;
+                let witness_hash = HashValue::from_slice(hash_bytes.as_slice())?;
+                let signature_bytes = deserializer.decode_bytes()?;
+                let witness_signature = Ed25519Signature::try_from(signature_bytes.as_slice())?;
```

Comment:
Ed25519Signature can use deserializer.decode_struct() to deserialize.

Reply:
updated